app/controllers/api_controller/reports.rb

```diff
@@ -44,5 +53,38 @@ def import_resource_reports(_type, _id, data)
       result, meta = MiqReport.import_from_hash(data["report"], options)
       action_result(meta[:level] == :info, meta[:message], :result => result)
     end
+
+    def schedule_resource_reports(_type, id, data)
+      schedule_data = fetch_schedule_data data
+      MiqReport.find(id).add_schedule schedule_data
```

Comment:
can fail, needed to rescue these and return the reason. I prefer the action_result() signatures.  Look at some of the vm methods, maybe reboot_guest_vm how the work is done and builds an action_result() based on the rescue. options for action_result can include :result if you want to return the schedule there, also on success we can return a "schedule_href" attribute as well as the parent (add_parent_href_to_result for the report href).